File: src/test/java/sds/offre/IdFormuleGenerateurDeInt.java
package sds.offre;

import sds.offre.concept_metier.IdFormule;
import sds.offre.concept_metier.IdFormuleGenerateur;

public class IdFormuleGenerateurDeInt implements IdFormuleGenerateur {
    private int valeur = 0;

    public IdFormule nouveauId() {
        return IdFormule.de(String.valueOf(++valeur));
    }
}

File: src/test/java/sds/offre/IdFormuleGenerateurDeInt.kt
package sds.offre

import sds.offre.concept_metier.IdFormule
import sds.offre.concept_metier.IdFormuleGenerateur

class IdFormuleGenerateurDeInt : IdFormuleGenerateur {
    private var valeur = 0

    override fun nouveauId(): IdFormule {
        return IdFormule((++valeur).toString())
    }
}
